src/policy/fees.cpp

```diff
@@ -76,116 +78,109 @@ void TxConfirmStats::UpdateMovingAverages()
     for (unsigned int j = 0; j < buckets.size(); j++)
     {
         for (unsigned int i = 0; i < confAvg.size(); i++)
+        {
             confAvg[i][j] = confAvg[i][j] * decay + curBlockConf[i][j];
+        }
         avg[j] = avg[j] * decay + curBlockVal[j];
-        txCtAvg[j] = txCtAvg[j] * decay + curBlockTxCt[j];
+        txCtAvg[j] = (txCtAvg[j] * decay) + curBlockTxCt[j];
     }
 }
 
 // returns -1 on error conditions
 double TxConfirmStats::EstimateMedianVal(int confTarget,
     double sufficientTxVal,
     double successBreakPoint,
-    bool requireGreater,
     unsigned int nBlockHeight)
 {
+    CAmount minTxFee = 0;
+    // its safe just to parse here because the validity of mintxfee was checked in init.cpp
+
+    // DEFAULT_TRANSACTION_MINFEE = 1000 defined in wallet.h, we hardcode it here as a hack fix
+    // due to --disable-wallet configs compiling policy estimator which would use wallet.h
+    // even though wallet.h isnt included when compiled with --disable-wallet
+    ParseMoney(std::to_string(GetArg("-mintxfee", 1000)), minTxFee);
+    // convert from satoshis to COIN
+    minTxFee = minTxFee / COIN;
```

Comment:
ParseMoney returns a value in Satoshis.  Dividing this by COIN (1E8) will yield 0 for every reasonable fee value.  Even BTC will hopefully never get to 1 BTC per KB!  Also, ParseMoney takes a value in COIN (BCH), so your default minfee is 1000 BCH

Reply:
fixd